.circleci/soltest_all.sh

```diff
@@ -28,10 +28,46 @@ set -e
 
 REPODIR="$(realpath $(dirname $0)/..)"
 
-EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 ${REPODIR}/.circleci/soltest.sh
+# This number has to equal the number of steps below.
+STEPS=11
+
+if [ "$CIRCLE_NODE_TOTAL" -a "$CIRCLE_NODE_TOTAL" -gt 1 ]
+then
+    # Run step 1 as the only step on the first executor
+    # and evenly distribute the other steps among
+    # the other executors.
+    # The first step takes much longer than the other steps.
+    if [ "$CIRCLE_NODE_INDEX" = "0" ]
+    then
+        RUN_STEPS="1"
+    else
+        export CIRCLE_NODE_INDEX=$(($CIRCLE_NODE_INDEX - 1))
+        export CIRCLE_NODE_TOTAL=$(($CIRCLE_NODE_TOTAL - 1))
+        RUN_STEPS=$(seq 2 "$STEPS" | circleci tests split)
+    fi
+else
+    RUN_STEPS=$(seq "$STEPS")
+fi
+
+# turn newlines into spaces
+RUN_STEPS=$(echo $RUN_STEPS)
+
+echo "Running steps $RUN_STEPS..."
+
+STEP=1
+
+[[ " $RUN_STEPS " =~ " $STEP " ]] && EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 ${REPODIR}/.circleci/soltest.sh
```

Comment:
Would be nice to make it work even if `$REPODIR` contains spaces: 
```suggestion
[[ " $RUN_STEPS " =~ " $STEP " ]] && EVM=istanbul OPTIMIZE=1 ABI_ENCODER_V2=1 "${REPODIR}/.circleci/soltest.sh"
```